include/chemfiles/files/MemoryBuffer.hpp

```diff
@@ -0,0 +1,66 @@
+// Chemfiles, a modern library for chemistry file reading and writing
+// Copyright (C) Guillaume Fraux and contributors -- BSD license
+
+#ifndef CHEMFILES_MEMORY_BUFFER_HPP
+#define CHEMFILES_MEMORY_BUFFER_HPP
+
+#include <vector>
+
+namespace chemfiles {
+
+class MemoryBuffer final {
```

Comment:
I am trying to avoid copying the `char*` buffer. If you have a different suggestion, I am all ears.

Reply:
The only thing I thought of is a custom allocator.

Reply:
> I am trying to avoid copying the char* buffer. If you have a different suggestion, I am all ears.

How about `span<char>`? It is designed exactly for this.

Reply:
The issue with `span<char>` is that it cannot be "grown" if needed.

Reply:
I may address this further in a future PR and this one is quite long!

In short, there would need to a version of `std::back_inserter` which operates on `span<char>` to grow it if needed. The `MemoryFile` class may need to be split in two again to support `span<char>` for writing and `span<const char>` for reading. However, I think this solution works for *now* and so I am marking as resolved.